iceoryx_binding_c/test/moduletests/test_notification_info.cpp

```diff
@@ -250,4 +256,162 @@ TEST_F(iox_notification_info_test, callbackCanBeCalledMultipleTimes)
     EXPECT_EQ(m_lastNotificationCallbackArgument, &m_userTrigger);
 }
 
+TEST_F(iox_notification_info_test, getNotificationInfoIdWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "d1c2e777-8166-426e-8a33-2193ca438d5f");
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { iox_notification_info_get_notification_id(nullptr); },
+                                              iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_notification_info_test, doesInfoOriginateFromSubscriberWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2a73ea5b-32c8-4e5a-9393-b10927d27b2a");
+    constexpr uint64_t ARBITRARY_EVENT_ID = 124U;
+    ASSERT_FALSE(m_waitSet.attachEvent(m_userTrigger, ARBITRARY_EVENT_ID).has_error());
+    m_userTrigger.trigger();
+
+    auto notificationInfoVector = m_waitSet.wait();
+
+    EXPECT_TRUE(iox_notification_info_does_originate_from_user_trigger(notificationInfoVector[0U], &m_userTrigger));
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_subscriber(nullptr, &m_subscriber); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_subscriber(notificationInfoVector[0U], nullptr); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_notification_info_test, doesInfoOriginateFromUserTriggerWithNullptrFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4ec226b6-0876-4ae4-94e0-c5ca91a12c98");
+    constexpr uint64_t ARBITRARY_EVENT_ID = 124U;
+    ASSERT_FALSE(m_waitSet.attachEvent(m_userTrigger, ARBITRARY_EVENT_ID).has_error());
+    m_userTrigger.trigger();
+
+    auto notificationInfoVector = m_waitSet.wait();
+
+    EXPECT_TRUE(iox_notification_info_does_originate_from_user_trigger(notificationInfoVector[0U], &m_userTrigger));
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_user_trigger(nullptr, &m_userTrigger); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>(
+        [&] { iox_notification_info_does_originate_from_user_trigger(notificationInfoVector[0U], nullptr); },
+        iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(iox_notification_info_test, doesOriginateFromClientWorks)
```

Comment:
@elBoberido The tests that tries to create a `client`, `server`, or a `service discovery` fails with exception, It seems that the test is not able to create an instance of those on the stack and gtest exits with exception. 
I am not sure what would be the best way to create these instances, can we share the test class from those modules with this module to create the instances, or how should it be done ?

Reply:
I had a quick look at `test_subscriber.cpp` and there it seems that the `RouDiEnvironment` should be used
```
iox::roudi::RouDiEnvironment roudiEnv;
iox_runtime_init("foo');
```
@elfenpiff can you confirm?

Reply:
Ouh ... I actually have no longer an idea. But I think I would try the roudi env first.

Reply:
@FerdinandSpitzschnueffler Thanks for your suggestion, I will give it a try today !